### PR TITLE
chore(deps): 🪦 deprecate enzyme 🪦

### PR DESCRIPTION
**Pull request by gkartalis**
This PR resolves []

### Description

This PR deprecates enzyme library in this repo 🪦.

Part of Refinathon, [ticketed here](https://www.notion.so/artsy/2673214ebb1f4693bf7b93f7d88cecee?v=a7c96c2dbf84404aad0c1bdaaa2e4902&p=b4efdc0fe1cd4f0682d3b4df9c73f628)

- [x] removes all helpers that we had for enzyme
- [x] removes enzyme
- [x] removes cheerio
- [x] removes @wojtekmaj/enzyme-adapter-react-17
- [x] removes @wojtekmaj/enzyme-adapter-react-17 patch
- [x] removes two hacks from `HACKS.md` related to enzyme

Related PRs before that (refactor enzyme tests to `@testing-library/react-native`):
- [x] https://github.com/artsy/eigen/pull/6978
- [x] https://github.com/artsy/eigen/pull/6996
- [x] https://github.com/artsy/eigen/pull/7003
- [x] https://github.com/artsy/eigen/pull/7006
- [x] https://github.com/artsy/eigen/pull/7011

Huge thanks to @dimatretyak 🌮 🎉 🚀  for moving this forwards!

### PR Checklist

- [ ] I tested my changes on **iOS** / **Android**.
- [ ] I added screenshots or videos to illustrate my changes.
- [x] I added Tests and Stories for my changes.
- [ ] I added an [app state migration].
- [ ] I hid my changes behind a [feature flag].

### To the reviewers 👀

- [ ] I would like **at least one** of the reviewers to run this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

-

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

- remove enzyme - gkartalis

</details>

Need help with something? Have a look at our [docs], or get in touch with us.

[app state migration]: ../blob/main/docs/adding_state_migrations.md
[feature flag]: ../blob/main/docs/developing_a_feature.md
[docs]: ../blob/main/docs/README.md

### CODE REVIEW

File: src/setupJest.ts [hunk 1]
```diff
@@ -1,19 +1,12 @@
 import "@testing-library/jest-native/extend-expect"
-import "jest-extended"
-
-import Adapter from "@wojtekmaj/enzyme-adapter-react-17"
 import { mockNavigate } from "app/tests/navigationMocks"
 import chalk from "chalk"
-// @ts-expect-error STRICTNESS_MIGRATION --- 🚨 Unsafe legacy code 🚨 Please delete this and fix any type errors if you have time 🙏
-import Enzyme from "enzyme"
 import expect from "expect"
+import "jest-extended"
```
Comment (dimatretyak) [inline src/setupJest.ts:5]: 🚀🚀🚀

File: src/setupJest.ts [hunk 2]
```diff
@@ -1,19 +1,12 @@
...
 import { format } from "util"
 
-import "app/tests/renderUntil"
-
 // MARK: - General preparation
 
-Enzyme.configure({ adapter: new Adapter() })
 const originalConsoleError = console.error
 // TODO: Remove once we're no longer using JSDOM for enzyme static rendering.
```
Comment (gkartalis) [inline src/setupJest.ts:11]: @pvinis do you think we should remove this weird log as well now?

**Review by dimatretyak — DISMISSED** [review]
🔥🔥🔥

**Review by dimatretyak — APPROVED** [review]

**Review by patrinoua — APPROVED** [review]

**Review by MounirDhahri — APPROVED** [review]
🎉

**Pull request merged**